Wikipedia/Code/PageHistoryCollectionViewCell.swift

```diff
@@ -52,8 +55,19 @@ class PageHistoryCollectionViewCell: CollectionViewCell {
         }
     }
 
+    func updateAccessibilityLabel() {
+        let isMinorAccessibilityString = isMinor ? WMFLocalizedString("page-history-revision-minor-edit-accessibility-label", value: "Minor edit", comment: "Accessibility label text used if edit was minor") : ""
+        accessibilityLabel = [timeLabel.accessibilityLabel, authorButton.accessibilityLabel, sizeDiffLabel.accessibilityLabel, isMinorAccessibilityString, commentLabel.accessibilityLabel]
+            .compactMap { $0 }
+            .filter { $0.wmf_hasNonWhitespaceText }
```

Comment:
Filter and compact map can be combined here into a single step

Reply:
True. Seemed cleaner this way than unwrapping optional in the filter... I don't feel strongly about it though. Also I may have missed a simpler approach 🤔

Reply:
Ah didn't think to move the logic into compactMap vs filter 😂...

Reply:
oh wait... seems similarly less clean...

Reply:
🤔

Reply:
Clean is subjective. I'd find it cleaner this way:
```
        .compactMap({ (maybeString) -> String? in
            guard
                let string = maybeString,
                string.wmf_hasNonWhitespaceText
            else {
                    return nil
            }
            return string
        })
```
Mainly because the types are explicit and it's easier to set breakpoints. In this case it doesn't really matter because you could inspect the entire list, but if it was a larger array it would be helpful to easily break on the nil condition and inspect what went wrong. Same goes for perf- the existing code iterates twice over the array, but again that doesn't really matter because the array has so few items.

Long story short, I don't feel strongly enough to hold up this PR.

Reply:
That's what I tried and thought was unjustifiably verbose given performance wasn't a concern and it was unlikely to be buggy, but perhaps I should have pulled it out into a function or declared it something like:

```
        let maybeNonWhitespaceString = { (maybeString: String?) -> String? in
            guard
                let string = maybeString,
                string.wmf_hasNonWhitespaceText
            else {
                    return nil
            }
            return string
        }
```

...then it's self-documenting at the point of use it reads nice and declaratively...

```            
.compactMap(maybeNonWhitespaceString)
```

...yet if issues are ever encountered you can dip into`maybeNonWhitespaceString` and set breakpoints, etc 🤔

Reply:
https://github.com/wikimedia/wikipedia-ios/pull/3375